core/genesis.go

```diff
@@ -349,6 +356,7 @@ func (g *Genesis) Commit(db ethdb.Database) (*types.Block, error) {
 	rawdb.WriteHeadBlockHash(db, block.Hash())
 	rawdb.WriteHeadHeaderHash(db, block.Hash())
 	rawdb.WriteChainConfig(db, block.Hash(), config)
+	rawdb.WriteUpgradeConfig(db, block.Hash(), &config.UpgradeConfig)
```

Comment:
probably can merge `WriteUpgradeConfig` and `WriteChainConfig`

Reply:
I think the reason to store UpgradeBytes is to ensure it is consistently provided (eg, so you can be sure of what upgrades you are running by examining `config.json` and `update.json`).

I think storing them under the same key complicates the serialization/deserialization.